registry.go

```diff
@@ -1,13 +1,6 @@
 package distribution
 
-import (
-	"io"
-	"time"
-
-	"github.com/docker/distribution/digest"
-	"github.com/docker/distribution/manifest"
-	"golang.org/x/net/context"
-)
+import "golang.org/x/net/context"
```

Comment:
I don't think the interface should rely on an external package like this. I see that it's used in the `Registry` interface below to get a repository, but wouldn't it make more sense for an implementation of `Registry` to decide if it's going to use `x/net/context`?